src/IntrType.f90

```diff
@@ -440,6 +567,78 @@ ELEMENTAL FUNCTION approxeq_rel_double(a,b) RESULT(bool)
 ENDFUNCTION approxeq_rel_double
 !
 !-------------------------------------------------------------------------------
+!> @brief Defines the operation for performing a relative comparison of one
+!> single-precision and one double-precision reals with .APPROXEQR.
+!> @param a
+!> @param b
+!> @returns bool logical indicating if a and b are approximately equal
+!>
+!> This performs a relative comparison by scaling the default epsilon value to
+!> the size of the larger of the two. It should be used when @c and @b are of
+!> the same magnitude and very large or very small. If either @c a or @c b is
+!> zero (exactly) then this routine is equivalent to an absolute comparison.
+!>
+ELEMENTAL FUNCTION approxeq_rel_mixsd(a,b) RESULT(bool)
+  REAL(SSK),INTENT(IN) :: a
+  REAL(SDK),INTENT(IN) :: b
+  LOGICAL(SBK) :: bool
+  REAL(SDK) :: eps
+  eps=MAX(ABS(a),ABS(b))*EPSS
+  IF(a == 0.0_SDK .OR. b == 0.0_SDK) eps=EPSS
+  bool=(ABS(a-b) <= eps)
+ENDFUNCTION approxeq_rel_mixsd
+!
+!-------------------------------------------------------------------------------
+!> @brief Defines the operation for performing a relative comparison of one
+!> double-precision and one single-precision reals with .APPROXEQR.
+!> @param a
+!> @param b
+!> @returns bool logical indicating if a and b are approximately equal
+!>
+!> This performs a relative comparison by scaling the default epsilon value to
+!> the size of the larger of the two. It should be used when @c and @b are of
+!> the same magnitude and very large or very small. If either @c a or @c b is
+!> zero (exactly) then this routine is equivalent to an absolute comparison.
+!>
+ELEMENTAL FUNCTION approxeq_rel_mixds(a,b) RESULT(bool)
+  REAL(SDK),INTENT(IN) :: a
+  REAL(SSK),INTENT(IN) :: b
+  LOGICAL(SBK) :: bool
+  REAL(SDK) :: eps
+  eps=MAX(ABS(a),ABS(b))*EPSS
+  IF(a == 0.0_SDK .OR. b == 0.0_SDK) eps=EPSS
+  bool=(ABS(a-b) <= eps)
+ENDFUNCTION approxeq_rel_mixds
+!
+!-------------------------------------------------------------------------------
+!> @brief Defines the operation for performing a relative comparison of two
+!> double-precision reals with .APPROXEQR. using single-precision criterion.
+!> @param a
+!> @param b
+!> @returns bool logical indicating if a and b are approximately equal
+!>
+!> This performs a relative comparison by scaling the default epsilon value to
+!> the size of the larger of the two. It should be used when @c and @b are of
+!> the same magnitude and very large or very small. If either @c a or @c b is
+!> zero (exactly) then this routine is equivalent to an absolute comparison.
+!>
+ELEMENTAL FUNCTION approxeq_rel_double2single(a,b) RESULT(bool)
```

Comment:
Strongly suggest removing this.

Reply:
The alternative is to use more scaling factors.